process/process.go

```diff
@@ -718,15 +718,21 @@ func (p *Process) insertPropose(propose Propose) bool {
 	// By never inserting a Propose that is not valid, we can avoid the validity
 	// checks elsewhere in the Process.
 	if p.validator != nil && !p.validator.Valid(propose.Value) {
+		// Broadcast a nil prevote for the proposal's round
 		if p.broadcaster != nil {
 			p.broadcaster.BroadcastPrevote(Prevote{
 				Height: p.CurrentHeight,
-				Round:  p.CurrentRound,
+				Round:  propose.Round,
 				Value:  NilValue,
 				From:   p.whoami,
 			})
+		}
+
+		// Step to prevoting if the proposal was for the round we are currently in
+		if p.CurrentRound == propose.Round {
 			p.stepToPrevoting()
 		}
+
```

Comment:
```suggestion
```
This is not needed. It will already be caught by `tryPrevoteUponPropose`.

Reply:
`tryPrevoteUponPropose` broadcasts a prevote only if it finds the propose in its `ProposeLogs`. In the above case, since the `validator.Valid(propose.Value)` says its invalid, we never insert it in the `ProposeLogs`. Hence the case was handled separately immediately.